floor_generator/lib/misc/extension/dart_object_extension.dart

```diff
@@ -9,10 +8,19 @@ extension DartObjectExtension on DartObject {
   String? toEnumValueString() {
     final interfaceType = type as InterfaceType;
     final enumName = interfaceType.getDisplayString(withNullability: false);
-    final enumValue = interfaceType.element.fields
+    final enumFields = interfaceType.element.fields
         .where((element) => element.isEnumConstant)
-        .map((fieldElement) => fieldElement.name)
-        .singleWhereOrNull((valueName) => getField(valueName) != null);
+        .toList();
+
+    // Find the index of the matching enum constant.
+    String? enumValue;
+    for (int i = 0; i < enumFields.length; i++) {
+      if (enumFields[i].computeConstantValue() == this) {
+        enumValue = OnConflictStrategy.values[i].name;
+        break;
+      }
+    }
```

Comment:
Hey, I would suggest replacing this construction with a separate private method for finding the index

```
  int? _getIndex(List<FieldElement> enumFields) => enumFields
        .asMap()
        .entries
        .firstWhereOrNull((e) => e.value.computeConstantValue() == this)
        ?.key;
```

This makes it easy to replace the cascade construction with

```
    final enumIndex = _getIndex(enumFields);
    final enumValue = enumIndex != null
            ? OnConflictStrategy.values[enumIndex].name
            : null;
```